src/plugins/auth.js

```diff
@@ -42,16 +42,18 @@ export const auth = {
             }
           }
 
-          const { preferred_username: preferredUsername } = user
+          const { oid } = user
 
-          if (!preferredUsername) {
-            logger.error('Authentication error: Missing preferred_username')
+          if (!oid || typeof oid !== 'string') {
```

Comment:
Do we always have `oid` set in our Microsoft flavoured tokens?

We can update all the JSDoc at the bottom of this file with the newer syntax either way

I've used `oid?: string` below so can get rid of this check and `'oid' in user` in **log-requests.js**

```js
/**
 * @import { ServerRegisterPluginObject } from '@hapi/hapi'
 * @import { HapiJwt } from '@hapi/jwt'
 * @import { IdTokenClaims } from 'oidc-client-ts'
 */

/**
 * @template {object} Payload
 * @typedef {HapiJwt.Artifacts<{ JwtPayload?: Payload }>} Artifacts
 */

/**
 * @typedef {IdTokenClaims & { groups?: string[], oid?: string }} UserProfile
 */
```

Reply:
```suggestion
          if (!user.oid) {
```

Reply:
Fixed per our alternative approach